src/Perspex.Controls/ControlValidationStatus.cs

```diff
@@ -0,0 +1,27 @@
+﻿using Perspex.Data;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Perspex.Controls
+{
+    public class ControlValidationStatus : ValidationStatus, INotifyPropertyChanged
```

Comment:
We'd need a listener for changing bindings to reset the validation state for each property whenever we change the binding. Otherwise we might lose validation checking if our binding changes and have the control marked as invalid when it is actually valid. I'll go work on that though.

We need the indexing by status because if we just track the last status we could have it marked invalid by Exception, but then marked valid by `INotifyDataErrorInfo` and then lose the validation information from Exception.